packages/block-editor/src/components/inserter/menu.native.js

```diff
@@ -125,48 +156,55 @@ export class InserterMenu extends Component {
 	}
 
 	render() {
-		const { items } = this.props;
 		const { numberOfColumns } = this.state;
 
 		return (
-			<BottomSheet
-				isVisible={ true }
-				onClose={ this.onClose }
-				hideHeader
-				hasNavigation
-			>
-				<TouchableHighlight accessible={ false }>
-					<BottomSheetConsumer>
-						{ ( { listProps, safeAreaBottomInset } ) => (
-							<FlatList
-								onLayout={ this.onLayout }
-								key={ `InserterUI-${ numberOfColumns }` } //re-render when numberOfColumns changes
-								keyboardShouldPersistTaps="always"
-								numColumns={ numberOfColumns }
-								data={ items }
-								ItemSeparatorComponent={ () => (
-									<TouchableWithoutFeedback
-										accessible={ false }
-									>
-										<View style={ styles.rowSeparator } />
-									</TouchableWithoutFeedback>
+			<ClipboardConsumer>
+				{ ( { clipboard } ) => (
```

Comment:
I wonder if this is the best approach as it's quite some boilerplate for just retrieving the clipboard. Also we don't really need to "react" on clipboard change here, the clipboard state will always be modified before opening the inserter.
Maybe a simple singleton object (like `isRTL` in `@wordpress/i18n`) would have been simpler to store clipboard data and allowed us to get the clipboard content outside a component's render method.
Any advantage I'm missing in using the Provider/Consumer approach?

Reply:
Thank you for your suggestion Tug! I updated the code to use a singleton approach instead.